### DISCUSSION TITLE
Home_load clarification

### DISCUSSION THREAD

**Discussion by eb99** [David00/rpi-power-monitor#147] [Q&A] [answered]
Hi,

Can anyone help me understand how the Home_load is calculated?
I have CTs on the grid connection (net) and solar production, and the hot water system.
The values of home_load seem too high, just by my gut really, feels like the hot water is added to the net so doubled up.
I was wondering if the home load is essentially adding up all CT values? or is it something else?
given my CT arrangement, I essentially have one CT measuring the Net value, so could we calculate raw ct values of net and solar to get the correct house consumption?
thanks in advance.

**Comment by David00** [comment, marked as answer]
Hi @eb99, the home_load is _intended_ to be the measurement of what your house is using in real time, regardless of _where_ the power is coming from, but it has some specific requirements to work properly.  Most notably, it requires measuring the incoming service mains with one CT per main feed. If you're not monitoring the mains, home_load is not going to be accurate.

The calculation of home_load depends on whether or not you have setup any channels with a `type = 'mains'` in your config.toml file.

If you do have `mains` channels:
* home_load is the sum of all mains channels + production channels.

If you do not have `mains` channels:
* home_load is the sum of all channels with `type = 'consumption'`

For the calculation to be accurate, it is imperative to ensure that the signage on the CT measurements is correct according to the Post Install Validation section of the docs here:
https://david00.github.io/rpi-power-monitor/docs/general/hardware-installation.html#post-install-validation

It is also crucial to have a sensor for each service main (in North America, we have two incoming mains, so you'd need one on each), and that the sensor for your solar inverter is properly configured in the config.toml file. 

The proper configuration of the CT on your solar inverter will vary, but for the following typical conditions:
* Using a single CT to measure solar output
* Solar inverter is grid-tied via a 2-pole circuit breaker

... then you'll want to set `two_pole = true` setting for this CT and ensure it's type is set to `production`.  

Hopefully this helps! Let me know if you have any additional questions :)

**Reply by eb99** [reply]
Thanks that answers it.

just another quick one, should a dashboard refresh be taking 30+s?

**Reply by David00** [reply]
I believe the default dashboard refresh in Grafana is 30 seconds. You can set a faster refresh time in the dropdown menu on the top right:

![image](https://github.com/user-attachments/assets/e5825698-1c28-49e3-905e-59ae337524b4)

**Comment by eb99** [comment]
hi,
no i mean the data takes the full 30s or longer to update, regardless of
the update rate.
i usually set i to 10min refresh, but that takes 30s to update the data.

**Reply by David00** [reply]
Oh, I see! Yes, that is normal.  Every 20-30 seconds, a new batch of data is flushed to the database, and the batch contains dozens of measurements about 1 second apart for those previous 20-30 seconds.
